### PR TITLE
Changed status code to follow conventions and some documentation fixes

### PR DESCRIPTION
**Pull request by Taucher2003**
I made some changes and will explain why I made them or what I made.

## Status codes

Some routes return the 200 status code which means OK but don't return any content. When a request returns no content they should use the 204 status code so it states that it returns no content.

This change may require some changes in the app if you expect the 200 status instead of the full 2xx range.

## Documentation

`server/core-services/CloudantConnection.js` removed a double space
`server/core-services/CloudantDb.js` changed rettry to retry
`server/core-services/DeviceTokenManager.js` changed it to is
`server/data-objects/EvaServiceData.js` changed hsitory to history

## Logging

In `server/server_app.js` some routes with a PUT mapping logged the requests as POST. This has been corrected.

### CODE REVIEW

**Review by Th3Ph4nt0m — APPROVED** [review]
Looks good!

File: server/core-services/DeviceTokenManager.js [hunk 1]
```diff
@@ -115,23 +115,23 @@ class DeviceTokenManager {
...
+      // If deletion fails this error is logged but device receives status 204 anyway. There
+      // is not much that can be done when deletion fails.
       if (await this.canRegister(grade, courseList, credential)) {
         this.logService.logger.info(`Updating device token ${deviceToken} for messaging provider ${messagingProvider} with grade ${grade} and course list [${courseList}]`);
 
         this.deviceTokensDb.get(deviceToken)
           .then(document => Object.assign(document, { _id: deviceToken, grade, courseList, messagingProvider, version, credential }))
           .then(newDocument => this.deviceTokensDb.insertDocument(newDocument))
-          .then(() => res.status(200).end())
+          .then(() => res.status(204).end())
```
Comment (michael-mosler) [inline server/core-services/DeviceTokenManager.js:126]: This is a breaking change that will cause app to stop working completely. Device tokens are registered with every restart of the app. When app receives another code than 200 at least iOS version is reporting this as an error.

I agree, 204 would be the correct response code. But with apps in the wild we need to introduce a new version of DeviceToken endpoint.

**Review by michael-mosler — CHANGES_REQUESTED** [review]
Sorry for not noticing this PR. I did not expect the Spanish Inquisition (small Monty Python insider). I fear that changing the response code in DeviceToken endpoint is a breaking change. See my comment below. In fact a version of the endpoint will be needed to correct this. We need to consider if it is worth the effort.

File: server/core-services/DeviceTokenManager.js (hunk 1 above)
Comment (Taucher2003) [reply server/core-services/DeviceTokenManager.js:126]: Then I will probably revert this change in favor of the current version of the apps

**Review by michael-mosler — CHANGES_REQUESTED** [review]
Sorry for getting back to you again. But there is a new code location in the same function which also now return 204 instead of 200. You also should update comment, please.

Thanks!
Michael

**Review by michael-mosler — APPROVED** [review]

**Pull request merged**